### ISSUE TITLE
PGP Release Key for Raspberry Pi is only 1024 bits. Please upgrade?

### ISSUE THREAD

**Issue by Danrancan** [meganz/MEGAcmd#569]
I am trying to add megacmd to my apt sources.list on a Raspberry Pi 4. 
I have installed it in accordance with my tutorial [https://nerd-tech.net/2021/09/06/how-to-install-and-update-mega-command-line-megacmd-on-your-raspberry-pi-running-ubuntu-20-04/], but noticed that the developer PGP release key is only 1024 bits. This is depricated. Is there a reason why the developers cannot produce a minimum of 2048 bit release key? If not, could you please work on this so I can properly verify my downloads on Raspbian OS? 

Thanks!

**Comment by AgentOak** [comment]
See also #402. No answer. I don't think security is something that MEGA cares about.

>Is there a reason why the developers cannot produce a minimum of 2048 bit release key

No, in fact 1024 bit weren't considered secure anymore since at least 2010. The fact that their 1024bit PGP key was generated on 2015-11-07 - a full five years after NIST officially discouraged use of RSA1024 - tells us that either:
- The developers are so negligent that the security of ALL MEGA products must be questioned
- They are actively malicious and deliberately inserted a backdoor

**Comment by polmr** [comment]
We will shortly update to a lengthier key. Thanks for your collaboration!

**Comment by Danrancan** [comment]
> We will shortly update to a lengthier key. Thanks for your collaboration!

Thank you so much. The current key is so outdated it is scary. Let me know when this has been accomplished if you can. You are welcome for the collaboration! Thanks for the response!

**Comment by Danrancan** [comment]
> We will shortly update to a lengthier key. Thanks for your collaboration!

Hey man, I apologize for sounding like an idiot, or an a-hole, but I have yet to see a new signing key from Mega.nz. This shouldn't take you guys more than 10 minutes to do. So its obsurd that this issue wasn't taken care of right away, as this is an URGENT MATTER for a company that claims to hold security in high regards. With only a 1024 bit key, why even bother having a key at all? Why even bother allowing the end user to verify downloads at all? Please guys, get on this as soon as possible before I start blogging about how irresponsible of a company Mega.nz is in terms of security, or even worse, question the overall integrity of your your company in public forums. I'm very much frowning upon you guys, but I still have hope. Please get on this as soon as possible! Even better, have it done before next year (January 1st, of 2021). Thanks, and sorry for kicking you with the boot in this post. Well... sorry not sorry, actually. Just trying to help understand priorities, so get on this!

**Comment by polmr** [comment]
Hi @Danrancan , 
This is now Work In Progress with high priority (it'll take more than 10 minutes though ;-) ) . Thanks for `kicking us in the arse` with this. Your concerns are really appreciated.

**Comment by Danrancan** [comment]
Here are some resources to help you quickly get this done...
[](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key)
[](https://wiki.debian.org/Keysigning)
and my personal recommendation...
[](https://github.com/drduh/YubiKey-Guide)

You have until February 28th, 2022 (A full month), before I write and spread many blog posts and forum posts about how MegaNZ is intentionally distributing insecure apps, potentially working with nation state governments, or even spying on its own customers for a schlew of possible reasons. I've asked you to uphold the bare minimum security standards here for a long time now, and you have had a long time to react and do something about it, yet, you've done nothing. Its only responsible and ethical to disclose irresponsibility (whether intentional or unintentional) to end users of your apps. Your lack of commitment makes it seem as if you are insanely negligent, or working with other entities to spy on end users and their data. I strongly urge you to take this time, to research and implement a strengthened signing key before I start spreading facts (not accusations) about our interactions and your negligence with your app security. It is unprofessional, and quite frankly, SCARY, how you have such an outdate algorithm being used in your pgp signatures. Please, this is my final warning. So for your sake and for mine, please do both ourselves a favor, and get your PGP signatures update before February 28th. Thats a full month of alotted time, and is an extremely fair timeline, so there is really no excuse for this not to be updated by then! Hope you get on this guys! 

Sincerely,
a concerned individual,
Dan Ran

**Comment by Danrancan** [comment]
> @polmr I'm sorry this message has to be targeted at you. It's actually meant for who ever is in charge of MegaNZ projects as a CEO or development head. So maybe you can relay this message to who ever this should go to...
> 
> 
> 
> First,
> 
> Here are some resources to help you quickly get this done...
> 
> 1) [https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key)
> 
> 
> 
> 2) [https://wiki.debian.org/Keysigning](https://wiki.debian.org/Keysigning)
> 
> and my personal recommendation...
> 
> 3) [https://github.com/drduh/YubiKey-Guide](https://github.com/drduh/YubiKey-Guide)
> 
> 
> 
> You have until February 28th, 2022 (A full month), before I write and spread many blog posts and forum posts about how MegaNZ is intentionally distributing insecure apps, potentially working with nation state governments, or even spying on its own customers for a schlew of possible reasons. I've asked you to uphold the bare minimum security standards here for a long time now, and you have had a long time to react and do something about it, yet, you've done nothing. Its only responsible and ethical to disclose irresponsibility (whether intentional or unintentional) to end users of your apps. Your lack of commitment makes it seem as if you are insanely negligent, or working with other entities to spy on end users and their data. I strongly urge you to take this time, to research and implement a strengthened signing key before I start spreading facts (not accusations) about our interactions and your negligence with your app security. It is unprofessional, and quite frankly, SCARY, how you have such an outdate algorithm being used in your pgp signatures. Please, this is my final warning. So for your sake and for mine, please do both ourselves a favor, and get your PGP signatures update before February 28th. Thats a full month of alotted time, and is an extremely fair timeline, so there is really no excuse for this not to be updated by then! Hope you get on this guys! 
> 
> 
> 
> Sincerely,
> 
> a concerned individual,
> 
> Dan Ran

paging @polmr.

**Comment by AgentOak** [comment]
It looks like MEGA has silently updated the key. The packages from https://mega.io/cmd now install the following RSA4096 key:

```
pub   rsa4096 2022-01-12 [SC] [expires: 2032-01-10]
      B01C811880480C854C73EC7E1A664B787094A482
uid           MegaLimited <support@mega.co.nz>
sub   rsa4096 2022-01-12 [E] [expires: 2032-01-10]
```

**But the new packages will not remove the old key - the dangerous RSA1024 key continues to be trusted if it was installed by an older version.**

I'm not sure if I have been clear enough before and if MEGA developers actually understand the issue at hand. You are installing this key in the system-wide APT trust store. This means the system will trust this key for *any package* from *any APT source*, not just the mega-cmd package. While the old key remains installed, the system is still at risk of total compromise, i.e. root-level execution once this flimsy RSA1024 key is cracked.

The package installer makes no attempt to find and remove the old key, nor did MEGA put out a warning to affected users whose systems continue to be vulnerable until the key is removed by hand. This is grossly negligent - and a pathetic excuse for a company that claims "MEGA is carefully engineered to achieve the highest level of [security](https://mega.io/security) for its users."

**Comment by Danrancan** [comment]
> The package installer makes no attempt to find and remove the old key, nor did MEGA put out a warning to affected users whose systems continue to be vulnerable until the key is removed by hand.

Agreed about this being negligent. But now that we have an update key, what is the problem with running 
`sudo apt-key del 8F208FBF12FEE766AA32AEAF03C3AD3A7F068E5D`, and then removing the sources.list.d file from your installation? Would this be a sufficient way of removing the vulnerability?

**Comment by polmr** [comment]
Hello there. We will remove the old key upon new installations. Thank you very much for your concerns.
